third_party/terraform/utils/retry_transport.go

```diff
@@ -202,6 +203,9 @@ func (t *retryTransport) checkForRetryableError(resp *http.Response, respErr err
 		errToCheck = googleapi.CheckResponse(&respToCheck)
 	}
 
+	if errToCheck == nil {
```

Comment:
nit: it feels a little safer to me if `isRetryableError` is responsible for the nil check.